src/api/app/jobs/report_to_scm_job.rb

```diff
@@ -17,12 +17,38 @@ def perform(event_id)
     EventSubscriptionsFinder.new
                             .for_scm_channel_with_token(event_type: event_type, event_package: event_package)
                             .each do |event_subscription|
-      SCMStatusReporter.new(event.payload,
-                            event_subscription.payload,
-                            event_subscription.token.scm_token,
-                            event_subscription.eventtype).call
+      workflow_filters = event_subscription.payload[:workflow_filters]
+
+      if report_for_repository?(event.payload['repository'], workflow_filters) && report_for_architecture?(event.payload['architecture'], workflow_filters)
```

Comment:
Looks like this could go into some `WorkflowFilter` object that `matches` an event using the new methods below and returns `true` or `false`. Doesn't it?